tests/test_connector.py

```diff
@@ -557,6 +557,40 @@ async def certificate_error(*args, **kwargs):
     assert isinstance(ctx.value, aiohttp.ClientSSLError)
 
 
+async def test_tcp_connector_server_hostname_default(loop) -> None:
+    conn = aiohttp.TCPConnector(loop=loop)
+
+    conn._loop.create_connection = mock.AsyncMock()
+    conn._loop.create_connection.return_value = mock.Mock(), mock.Mock()
```

Comment:
Would be great if you can get this to work with something like:

```suggestion
    with mock.patch.object(conn._loop, "create_connection", autospec=True, spec_set=True):
```